src/reducers/ThemeSlice.ts

```diff
@@ -0,0 +1,19 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+const initialState = {
+  mode: localStorage.getItem('theme') || 'light',
```

Comment:
Please forget about this "other suggestion" for now, because this needs to change all `useAppSelector` for the theme mode everywhere, and I'd rather move this PR forward than do that.
In the future we can encapsulate the "preferedMode computation" in a hook, but that's out of scope.